lib/api/openid_connect/token_endpoint.rb

```diff
@@ -0,0 +1,69 @@
+module Api
+  module OpenidConnect
+    class TokenEndpoint
+      attr_accessor :app
+      delegate :call, to: :app
+
+      def initialize
+        @app = Rack::OAuth2::Server::Token.new do |req, res|
+          o_auth_app = retrieve_client(req)
+          if app_valid?(o_auth_app, req)
+            handle_flows(o_auth_app, req, res)
+          else
+            req.invalid_client!
+          end
+        end
+      end
+
+      def handle_flows(o_auth_app, req, res)
+        case req.grant_type
+        when :password
+          handle_password_flow(o_auth_app, req, res)
+        when :refresh_token
+          handle_refresh_flow(req, res)
+        else
+          req.unsupported_grant_type!
+        end
+      end
+
+      def handle_password_flow(o_auth_app, req, res)
```

Comment:
Assignment Branch Condition size for handle_password_flow is too high. [16.55/15]